frontend/public/components/pod.tsx

```diff
@@ -304,136 +304,127 @@ const getSelectedColumns = (showNodes: boolean) => {
-          ? new Set(tableColumns[columnManagementID])
-          : getSelectedColumns(showNodes);
-      return (
-        <TableRow id={pod.metadata.uid} index={index} trKey={rowKey} style={style}>
-          <TableData className={podColumnInfo.name.classes}>
-            <ResourceLink kind={kind} name={name} namespace={namespace} />
-          </TableData>
-          <TableData
-            className={classNames(podColumnInfo.namespace.classes, 'co-break-word')}
-            columns={columns}
-            columnID={podColumnInfo.namespace.id}
-          >
-            <ResourceLink kind="Namespace" name={namespace} />
-          </TableData>
-          <TableData
-            className={podColumnInfo.status.classes}
-            columns={columns}
-            columnID={podColumnInfo.status.id}
-          >
-            <PodStatus pod={pod} />
-          </TableData>
-          <TableData
-            className={podColumnInfo.ready.classes}
-            columns={columns}
-            columnID={podColumnInfo.ready.id}
-          >
-            {readyCount}/{totalContainers}
-          </TableData>
-          <TableData
-            className={podColumnInfo.restarts.classes}
-            columns={columns}
-            columnID={podColumnInfo.restarts.id}
-          >
-            {restarts}
-          </TableData>
-          <TableData
-            className={podColumnInfo.owner.classes}
-            columns={columns}
-            columnID={podColumnInfo.owner.id}
-          >
-            {showNodes ? (
-              <ResourceLink kind="Node" name={pod.spec.nodeName} namespace={namespace} />
-            ) : (
-              <OwnerReferences resource={pod} />
-            )}
-          </TableData>
-          <TableData
-            className={podColumnInfo.memory.classes}
-            columns={columns}
-            columnID={podColumnInfo.memory.id}
-          >
-            {bytes ? `${formatBytesAsMiB(bytes)} MiB` : '-'}
-          </TableData>
-          <TableData
-            className={podColumnInfo.cpu.classes}
-            columns={columns}
-            columnID={podColumnInfo.cpu.id}
-          >
-            {cores ? `${formatCores(cores)} cores` : '-'}
-          </TableData>
-          <TableData
-            className={podColumnInfo.created.classes}
-            columns={columns}
-            columnID={podColumnInfo.created.id}
-          >
-            <Timestamp timestamp={creationTimestamp} />
-          </TableData>
-          <TableData
-            className={podColumnInfo.node.classes}
-            columns={columns}
-            columnID={podColumnInfo.node.id}
-          >
+  ({
+    obj: pod,
+    index,
+    rowKey,
+    style,
+    metrics,
+    showNodes,
+  }: PodTableRowProps & PodTableRowPropsFromState) => {
+    const [tableColumns, , loaded] = useUserSettingsCompatibility(
+      COLUMN_MANAGEMENT_CONFIGMAP_KEY,
+      COLUMN_MANAGEMENT_LOCAL_STORAGE_KEY,
+      null,
```

Comment:
Please keep the default as undefined so that this value are not stored in user settings configmap until the user changes the table config.